### PR TITLE
task_DEV-3

### PR DESCRIPTION
**Pull request by YauheniLoban**

### CODE REVIEW

File: HomeTask3/HomeTask3/Program.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            //We check the arguments for the correctness of the input (only numbers can be entered)
+            if(args[0] == string.Empty || args[1] == string.Empty)
+            {
+                Console.WriteLine("You entered an empty string");
+                return;
+
```
Comment (IgorKheidorov) [inline HomeTask3/HomeTask3/Program.cs]: Code style

File: HomeTask3/HomeTask3/Program.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            {
+                Console.WriteLine("You entered not only numbers in the argument string. Please correct the arguments");
+                return ;
+            }
+            if (Convert.ToInt16(args[1]) < 2 || Convert.ToInt16(args[1]) > 20)
+            {
+                Console.WriteLine("You entered the base of the new system not from the allowed range. Please enter another base of the new system ");
+                return ;
+            }
+            translationIntoNewSystem newConvertedString = new translationIntoNewSystem(args);
```
Comment (IgorKheidorov) [inline HomeTask3/HomeTask3/Program.cs]: Code style

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Text;
+
+namespace DevTask3
+{
+    /// <summary> 
+    /// translationIntoNewSystem
```
Comment (IgorKheidorov) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: Code style

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            figureInNewSystem = possibleValuesOfElementsInNewView[index];
+            return Convert.ToString(figureInNewSystem);
+        }
+        /// <summary> 
+        /// reverseString
+        /// Function to create a string of characters in which go in reverse order
+        /// </summary> 
+        /// <param name="stringToReverse">The string you need to revers</param> 
+        /// <returns>Reversed String</returns>
+        private string reverseString(string stringToReverse)
```
Comment (IgorKheidorov) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: code style

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs (hunk 4 above)
Comment (YauheniLoban) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs]: too many times used String?

File: HomeTask3/HomeTask3/Program.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    /// <summary> 
+    /// We accept from the command line two arguments: 
+    /// 1) the number that needs to be transferred to another system of calculation 
+    /// 2) the foundation of a new system of calculus 
+    /// </summary> 
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            // We check the arguments for the correctness of the input (only numbers can be entered)
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/Program.cs]: useless comment as it's wrong

File: HomeTask3/HomeTask3/Program.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                Console.WriteLine("You entered an empty string");
+                return;
+            }
+            try 
+            {
+                long digitForTranslation;
+                long foundationOfNewsystem;
+                digitForTranslation = Int64.Parse(args[0]);
+                foundationOfNewsystem = Int64.Parse(args[1]);
+            }
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/Program.cs]: wrong exception usage

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Text;
+
+namespace DevTask3
+{
+    /// <summary> 
+    /// This class creates a string that is a translation of a number into another number system
+    /// </summary>
+    public class TranslationIntoNewSystem
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: Too complicated class name

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        private long numberForConversion;
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: why long?

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        private long foundationOfNewSystem;
+
+        public TranslationIntoNewSystem(string[] valuesNumberAndBaseOfSystem)
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: pass two arguments instead error-prone array

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        public string CreatingRepresentationOfNumberInNewSystem()
+        {
+            StringBuilder valueInNewSystem = new StringBuilder();
+            long nextNumberToAdd;
+            do
+            {
+                nextNumberToAdd = numberForConversion % foundationOfNewSystem;
+                valueInNewSystem.Append(ConvertingFigureToRepresentationInAnotherSystem(nextNumberToAdd));
+                numberForConversion = numberForConversion / foundationOfNewSystem;
+            } while (0 != numberForConversion);
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: check in codestyle whether while should be on new line or not

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+
+        /// <summary> 
+        /// Method ConvertingFigureToRepresentationInAnotherSystem
+        /// Function for converting numbers to a new number system
+        /// </summary> 
+        /// <param name="digitForTranslation">number to convert to a new system</param> 
+        /// <returns>Figure In New System</returns>
+        private string ConvertingFigureToRepresentationInAnotherSystem(long digitForTranslation)
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: private methods don't need to be commented

Comment (AlexanderSoroka) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs]: ExtremlyLongNameForSuchSimpleMethod

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+        {
+            char figureInNewSystem;
+            int index;
+            index = Convert.ToInt16(digitForTranslation);
+            var possibleValuesOfElementsInNewView = "0123456789ABCDEFGHIJ";
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs]: must be const, avoid using var just because you don't know the type )

Comment (AlexanderSoroka) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs]: use shorter and clearer names

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+        /// Method ReverseString
+        /// Function to create a string of characters in which go in reverse order
+        /// </summary> 
+        /// <param name="stringToReverse">The string you need to revers</param> 
+        /// <returns>Reversed String</returns>
+        private string ReverseString(string stringToReverse)
+        {
+            char[] reversedString = stringToReverse.ToCharArray();
+            Array.Reverse(reversedString);
+            return new string(reversedString);
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/translationIntoNewSystem.cs:52]: I would write everything in 1 line

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs (hunk 8 above)
Comment (YauheniLoban) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs]: Someone said that it's better to use a type that can accommodate more than int. To remake under int ?

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs (hunk 10 above)
Comment (YauheniLoban) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs]: image
should not be on a new line

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs (hunk 13 above)
Comment (YauheniLoban) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs:52]: You mean that we can implement it more easily?

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs (hunk 8 above)
Comment (AlexanderSoroka) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs]: https://msdn.microsoft.com/en-us/library/296az74e.aspx?f=255&MSPPError=-2147217396
:D
You'll be surprised. Always read documentation before doing stupid things

File: HomeTask3/HomeTask3/Program.cs (hunk 6 above)
Comment (YauheniLoban) [reply HomeTask3/HomeTask3/Program.cs]: Could you clarify. I don't quite understand the error,

File: HomeTask3/HomeTask3/Program.cs [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    /// 1) the number that needs to be transferred to another system of calculation 
+    /// 2) the foundation of a new system of calculus 
+    /// </summary> 
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            try 
+            {
+                int digitForTranslation;
```
Comment (AlexanderSoroka) [inline HomeTask3/HomeTask3/Program.cs:17]: `int digitForTranslation = Int32.Parse(args[0]);`

File: HomeTask3/HomeTask3/translationIntoNewSystem.cs (hunk 13 above)
Comment (AlexanderSoroka) [reply HomeTask3/HomeTask3/translationIntoNewSystem.cs:52]: Nope, just rewrite it without tmp variables

**Review by AlexanderSoroka — APPROVED** [review]

**Pull request merged**